tests/autorestart_test/test_container_autorestart.py

```diff
@@ -0,0 +1,149 @@
+    """
+    is_running = duthost.shell("docker inspect -f \{\{.State.Running\}\} %s" % container_name)
+    if is_running['stdout'].strip() == "false" and should_be_stopped:
+        return True
+    if is_running['stdout'].strip() == "true" and not should_be_stopped:
+        return True
+    return False
+ 
+def verify_autorestart_with_critical_process(duthost, container_name, process):
+    """
+    @summary: Killing a critical process in a container to verify whether the container 
+              can be stopped and then restarted correctly 
+    """
+    kill_process(duthost, container_name, process)
+
+    logging.info("Waiting until {} is stopped...".format(container_name))
+    assert wait_until(CONTAINER_STOP_THRESHOLD_SECS, 
+                      CONTAINER_CHECK_INTERVAL_SECS,
+                      check_container_status, duthost, container_name, True)
+    logging.info("{} is stopped".format(container_name))
+
+    logging.info("Waiting until {} is restarted...".format(container_name))
+    assert wait_until(CONTAINER_RESTART_THRESHOLD_SECS, 
+                      CONTAINER_CHECK_INTERVAL_SECS,
+                      check_container_status, duthost, container_name, False)
+    logging.info("{} is restarted".format(container_name))
+
+def verify_autorestart_with_non_critical_process(duthost, container_name, process):
+    """
+    @summary: Killing a non-critical process in a container to verify whether the container 
+              is still in the running state 
+    """
+    kill_process(duthost, container_name, process)
+
+    logging.info("Checking whether the {} is still running...".format(container_name))
+    assert wait_until(CONTAINER_STOP_THRESHOLD_SECS, 
+                      CONTAINER_CHECK_INTERVAL_SECS,
+                      check_container_status, duthost, container_name, False)
+    logging.info("{} is running".format(container_name))
+
+@pytest.fixture(scope="module", autouse=True)
+def change_autorestart_status(duthost):
+    container_list = ["lldp", "radv", "pmon", "sflow", "database", "telemetry", "snmp",
+                      "bgp", "dhcp_relay", "nat", "teamd", "syncd", "swss"]
+    for container_name in container_list:
+        logging.info("Change {} auto-restart state to enabled".format(container_name))
+        duthost.shell("config container feature autorestart %s enabled" % container_name)
+
+def test_swss_autorestart(duthost):
+    verify_autorestart_with_critical_process(duthost, "swss", "portsyncd")
+    verify_autorestart_with_non_critical_process(duthost, "swss", "rsyslogd")
+
+def test_lldp_autorestart(duthost):
+    verify_autorestart_with_critical_process(duthost, "lldp", "lldpmgrd")
+    verify_autorestart_with_non_critical_process(duthost, "lldp", "rsyslogd")
+
+def test_sflow_autorestart(duthost):
+    verify_autorestart_with_critical_process(duthost, "sflow", "sflowmgrd")
+    verify_autorestart_with_non_critical_process(duthost, "sflow", "rsyslogd")
```

Comment:
While I like the idea of having a separate test for each container, I'm concerned about maintaining this in the future (e.g., when we add/remove containers). Another option is to automate this by getting the list of containers which have the autorestart feature by checking the output of `show container feature autorestart`, and looping through that list. I think this is worth discussing pros and cons.

Reply:
I agree with this method to first get the container list which has the autorestart feature and then test each container one by one. In this way, it is clean and also easy to maintain but we have to test all the containers in a single testing function.

Reply:
Yes, there is a tradeoff. @yxieca, @lguohan: Do you have an opinion on my proposal?